### PR TITLE
[SPARK-31204][SQL] HiveResult compatibility for DatasourceV2 command

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

`HiveResult` performs some conversions for commands to be compatible with Hive output, e.g.:
```
// If it is a describe command for a Hive table, we want to have the output format be similar with Hive.
case ExecutedCommandExec(_: DescribeCommandBase) =>
...
// SHOW TABLES in Hive only output table names, while ours output database, table name, isTemp.
case command @ ExecutedCommandExec(s: ShowTablesCommand) if !s.isExtended =>
```
This conversion is needed for DatasourceV2 commands as well and this PR proposes to add the conversion for v2 commands `SHOW TABLES` and `DESCRIBE TABLE`.

### Why are the changes needed?

This is a bug where conversion is not applied to v2 commands.

### Does this PR introduce any user-facing change?

Yes, now the outputs for v2 commands `SHOW TABLES` and `DESCRIBE TABLE` are compatible with HIVE output.

For example, with a table created as:
```
CREATE TABLE testcat.ns.tbl (id bigint COMMENT 'col1') USING foo
```

The output of `SHOW TABLES` has changed from
```
ns    table
```
to
```
table
```

And the output of `DESCRIBE TABLE` has changed from
```
id    bigint    col1

# Partitioning        
Not partitioned        
```
to
```
id                      bigint                  col1                
                                                                           
# Partitioning                                                    
Not partitioned                                                 
```

### How was this patch tested?

Added unit tests.

### CODE REVIEW

**Comment by imback82** [comment]
cc @cloud-fan @juliuszsompolski

**Comment by HyukjinKwon** [comment]
> Yes, now the outputs for v2 commands SHOW TABLES and DESCRIBE TABLE are compatible with HIVE output.

Can you show before/after input/output?

**Comment by imback82** [comment]
> Can you show before/after input/output?

I updated the description. Thanks.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/HiveResultSuite.scala [hunk 1]
```diff
@@ -68,4 +69,35 @@ class HiveResultSuite extends SharedSparkSession {
...
+      Seq(("testcat.ns", "tbl", "foo"), ("spark_catalog.default", "tbl", "csv")).foreach {
+        case (ns, tbl, source) =>
+          withTable(s"$ns.$tbl") {
+            spark.sql(s"CREATE TABLE $ns.$tbl (id bigint COMMENT 'col1') USING $source")
+            val df = spark.sql(s"DESCRIBE $ns.$tbl")
+            val executedPlan = df.queryExecution.executedPlan
+            val expected = "id                  " +
+              "\tbigint              " +
+              "\tcol1                "
+            assert(HiveResult.hiveResultString(executedPlan).head == expected)
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/HiveResultSuite.scala]: can we make the test more robust? e.g.
```
assert(HiveResult.hiveResultString(executedPlan).head.split("\n").map(_.trim) == Seq("id", "bigint", "col1"))
```

**Comment by cloud-fan** [comment]
LGTM, can we fix the conflicts?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/HiveResultSuite.scala (hunk 1 above)
Comment (imback82) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/HiveResultSuite.scala]: If you `trim`, wouldn't it make the test less robust? Adding the padding was the compatibility change for `DESCRIBE`.

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/HiveResultSuite.scala]: ok so the number of spaces is also defined bu the Hive.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**